inference-engine/tests/functional/plugin/shared/src/single_layer_tests/interpolate.cpp

```diff
@@ -0,0 +1,94 @@
+
+#include "single_layer_tests/interpolate.hpp"
+
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string mode2String(ngraph::op::Interpolate::InterpolateMode mode) {
+    switch (mode) {
+        case ngraph::op::Interpolate::InterpolateMode::area:
+            return "area";
+        case ngraph::op::Interpolate::InterpolateMode::cubic:
+            return "cubic";
+        case ngraph::op::Interpolate::InterpolateMode::linear:
+            return "linear";
+        case ngraph::op::Interpolate::InterpolateMode::nearest:
+            return "nearest";
+        default:
+            throw std::runtime_error("NOT_SUPPORTED_MODE_TYPE");
+    }
+}
+
+std::string InterpolateLayerTest::getTestCaseName(testing::TestParamInfo<InterpolateLayerTestParams> obj) {
+    InterpolateSpecificParams interpolateParams;
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::SizeVector inputShapes, targetShapes;
+    std::string targetDevice;
+    std::tie(interpolateParams, netPrecision, inputShapes, targetShapes, targetDevice) = obj.param;
+    std::set<size_t> axes;
+    std::vector<size_t> padBegin, padEnd;
+    bool antialias, alignCorners;
+    ngraph::op::Interpolate::InterpolateMode mode;
+    std:tie(axes, mode, alignCorners, antialias, padBegin, padEnd) = interpolateParams;
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    result << "TS=" << CommonTestUtils::vec2str(targetShapes) << "_";
+    result << "Axe=" << CommonTestUtils::set2str(axes)<< "_";
+    result << "Mode=" << mode2String(mode) << "_";
+    result << "AlignCorners=" << alignCorners << "_";
+    result << "Antialias=" << antialias << "_";
+    result << "PB=" << CommonTestUtils::vec2str(padBegin) << "_";
+    result << "PE=" << CommonTestUtils::vec2str(padEnd) << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "targetDevice=" << targetDevice;
+    return result.str();
+}
+
+void InterpolateLayerTest::SetUp() {
+    InterpolateSpecificParams interpolateParams;
+    std::vector<size_t> inputShape, targetShape;
+    auto netPrecision = InferenceEngine::Precision::UNSPECIFIED;
+
+    std::tie(interpolateParams, netPrecision, inputShape, targetShape, targetDevice) = this->GetParam();
+    std::set<size_t> axes;
+    std::vector<size_t> padBegin, padEnd;
+    bool antialias, alignCorners;
+    ngraph::op::Interpolate::InterpolateMode mode;
+    std::tie(axes, mode, alignCorners, antialias, padBegin, padEnd) = interpolateParams;
+
+    if (targetShape.size() != axes.size()) {
+        THROW_IE_EXCEPTION << "Target shape size: " << targetShape.size() << " is not equal Axes shapes: " <<  axes.size();
+    }
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+    auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+    std::vector<float> data(targetShape.size());
```

Comment:
this copying looks strange for me. Is it done just to fit makeConstant interface?

I propose to use `Constant(const element::Type& type, Shape shape, const std::vector<T>& values)` constructor diretcly and get rid of creating additional vector `data`

Reply:
Fixed